### PR TITLE
Abstract over different HttpClient implementations

### LINKED ISSUE #23: Support for user-injected hyper client
Hi!

What are your thoughts on supporting a user-injected hyper client? I'm referring to the client used [here](https://github.com/instant-labs/instant-acme/blob/main/src/lib.rs#L350).

### PR DESCRIPTION
**Pull request by djc**
Fixes #23.

### CODE REVIEW

**Comment by rikonor** [comment]
Wow! Thanks so much for going to the trouble, @djc.

**Review by rikonor — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -533,15 +593,36 @@ fn nonce_from_response(rsp: &Response<Body>) -> Option<String> {
...
+                    .enable_http2()
+                    .build(),
+            ),
+        )
+    }
+}
 
-    hyper::Client::builder().build(https)
+/// A HTTP client based on [`hyper::Client`]
+pub trait HttpClient {
```
Comment (Daniel-Bloom-dfinity) [inline src/lib.rs:623]: Can you add an `impl` for all `Client`s?

```rust
impl<C> HttpClient for hyper::Client<C> where
    C: hyper::client::connect::Connect + Clone + Send + Sync + 'static {
    fn request(&self, req: Request<Body>) -> ResponseFuture {
        <hyper::Client<C>>::request(self, req)
    }
}
```

Comment (djc) [reply src/lib.rs:624]: Good idea, done.

**Comment by djc** [comment]
Published as 0.3.1.

**Pull request merged**